vizro-core/src/vizro/models/_components/figure.py

```diff
@@ -0,0 +1,63 @@
+from typing import Literal, Optional
+
+import pandas as pd
+from dash import dcc, html
+
+try:
+    from pydantic.v1 import Field, PrivateAttr, validator
+except ImportError:  # pragma: no cov
+    from pydantic import Field, PrivateAttr, validator
+
+import vizro.figures as vf
+from vizro.managers import data_manager
+from vizro.models import VizroBaseModel
+from vizro.models._components._components_utils import _callable_mode_validator_factory, _process_callable_data_frame
+from vizro.models._models_utils import _log_call
+from vizro.models.types import CapturedCallable
+
+
```

Comment:
@antonymilne - given that I already know there will be a big discussion. Could you just let me know which names you absolutely dislike? 😄